File: Source/Fantasy.Jobs/Resources/ResourceRequestQueue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.ServiceModel;

namespace Fantasy.Jobs.Resources
{
    public class ResourceRequestQueue : AbstractService, IResourceRequestQueue
    {

        private Dictionary<Guid, ResourceParameter[]> _queue = new Dictionary<Guid, ResourceParameter[]>();

        private object _syncRoot = new object();
        

        public void RegisterResourceRequest(Guid jobId, ResourceParameter[] parameters)
        {
            lock (_syncRoot)
            {
                if (!_queue.ContainsKey(jobId))
                {
                    _queue.Add(jobId, parameters);
                }
            }
        }

        public void UnregisterResourceRequest(Guid jobId)
        {
            lock (_syncRoot)
            {
                this._queue.Remove(jobId);
            }
        }



        #region IResourceRequestQueue Members


        public ResourceParameter[] GetRequiredResources(Guid jobId)
        {
            ResourceParameter[] rs;
            lock (_syncRoot)
            {
                rs = this._queue.GetValueOrDefault(jobId, new ResourceParameter[] { });  
            }
            return rs;
        }

        #endregion
    }
}

File: AutoConverted/Fantasy.Jobs/Resources/ResourceRequestQueue.java
package Fantasy.Jobs.Resources;

import Fantasy.ServiceModel.*;

public class ResourceRequestQueue extends AbstractService implements IResourceRequestQueue
{

	private java.util.HashMap<Guid, ResourceParameter[]> _queue = new java.util.HashMap<Guid, ResourceParameter[]>();

	private Object _syncRoot = new Object();


	public final void RegisterResourceRequest(Guid jobId, ResourceParameter[] parameters)
	{
		synchronized (_syncRoot)
		{
			if (!_queue.containsKey(jobId))
			{
				_queue.put(jobId, parameters);
			}
		}
	}

	public final void UnregisterResourceRequest(Guid jobId)
	{
		synchronized (_syncRoot)
		{
			this._queue.remove(jobId);
		}
	}



//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region IResourceRequestQueue Members


	public final ResourceParameter[] GetRequiredResources(Guid jobId)
	{
		ResourceParameter[] rs;
		synchronized (_syncRoot)
		{
			rs = this._queue.GetValueOrDefault(jobId, new ResourceParameter[] { });
		}
		return rs;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}